### ISSUE TITLE
rm does not work with variable (no such file or directory)

### ISSUE THREAD

**Issue by doongjohn** [nushell/nushell#11061]
### Describe the bug

```nu
cd ~/apps/neovim

print -n '[script] removing all files in "' $env.PWD "\"\n"
rm -fr ./*

print '[script] downloading'
let url = (
  http get https://api.github.com/repos/neovim/neovim/releases
  | first
  | get assets.browser_download_url
  | find --regex win64.zip$
  | first)

let zip_file_name = 'neovim-win64.zip'
http get $url | save -p ('' | path join $env.PWD $zip_file_name)

print '[script] extracting'
7z x $zip_file_name
rm neovim-win64.zip # <-- this works
# rm $zip_file_name # <-- this does not work (no such file or directory)

mv 'nvim-win64/*' './'
rm -r 'nvim-win64/'

print '[script] complete'
```
I'm using this script to update neovim.
However It does not work after I updated nushell to 0.87.0

### How to reproduce

1. comment `rm neovim-win64.zip` and uncomment `rm $zip_file_name` 
2. run the example script

### Expected behavior

no error

### Screenshots

![image](https://github.com/nushell/nushell/assets/40219740/d04a2728-f680-4c70-82e6-387a2db8eee8)

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.87.0                                   |
| branch             |                                          |
| commit_hash        | 77a1c3c7b2f3a110d48bcb792968e6b0d85d4bb7 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.71.1 (eb26296b5 2023-08-03)      |
| rust_channel       | 1.71.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.71.1 (7f1d04c00 2023-07-29)      |
| build_time         | 2023-11-14 21:01:56 +00:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash, which, zip       |
| installed_plugins  |                                          |

**Comment by fdncred** [comment]
Does it make any difference if you do the remove like this?
```nushell
rm ($zip_file_name)
```

**Comment by doongjohn** [comment]
![image](https://github.com/nushell/nushell/assets/40219740/0025c665-0d2c-4e7e-938d-0be992f24f49)
it still give me an error

**Comment by fdncred** [comment]
hmmm, maybe `rm` doesn't understand variables yet?

**Comment by doongjohn** [comment]
but this works as expected
```nu
let file = 'test.txt'
touch $file
rm $file
```

**Comment by IanManske** [comment]
This looks related to #10832. If you do `rm ($zip_file_name | path expand)` does that work?

**Comment by doongjohn** [comment]
`rm ($zip_file_name | path expand)` this works!

**Comment by IanManske** [comment]
Yeah, that's weird. I don't know why one works but the other doesn't.

**Comment by doongjohn** [comment]
Fixed in `0.87.1`
Thanks for the hotfix!
